fast64_internal/f3d/f3d_material.py

```diff
@@ -177,85 +177,144 @@ def update_draw_layer(self, context):
+        return default_for_no_rendermode
+    is_one_cycle = settings.g_mdsft_cycletype == "G_CYC_1CYCLE"
+    return (
+        settings.blend_p1 == color or settings.blend_m1 == color
+        or (is_one_cycle and (settings.blend_p2 == color or settings.blend_m2 == color))
+    )
+    
 
-    is_one_cycle = f3dMat.rdp_settings.g_mdsft_cycletype == "G_CYC_1CYCLE"
+def does_blender_use_alpha(
+    settings: "RDPSettings",
+    alpha: str,
+    default_for_no_rendermode: bool = False
+) -> bool:
+    if not settings.set_rendermode:
+        return default_for_no_rendermode
+    is_one_cycle = settings.g_mdsft_cycletype == "G_CYC_1CYCLE"
+    return settings.blend_a1 == alpha or (is_one_cycle and settings.blend_a2 == alpha)
 
-    if f3dMat.rdp_settings.set_rendermode:
-        if f3dMat.rdp_settings.rendermode_advanced_enabled:
-            if f3dMat.rdp_settings.cvg_x_alpha:
-                blend_method = "CLIP"
-            elif (
-                is_one_cycle
-                and f3dMat.rdp_settings.force_bl
-                and f3dMat.rdp_settings.blend_p1 == "G_BL_CLR_IN"
-                and f3dMat.rdp_settings.blend_a1 == "G_BL_A_IN"
-                and f3dMat.rdp_settings.blend_m1 == "G_BL_CLR_MEM"
-                and f3dMat.rdp_settings.blend_b1 == "G_BL_1MA"
-            ):
-                blend_method = "BLEND"
-            elif (
-                not is_one_cycle
-                and f3dMat.rdp_settings.force_bl
-                and f3dMat.rdp_settings.blend_p2 == "G_BL_CLR_IN"
-                and f3dMat.rdp_settings.blend_a2 == "G_BL_A_IN"
-                and f3dMat.rdp_settings.blend_m2 == "G_BL_CLR_MEM"
-                and f3dMat.rdp_settings.blend_b2 == "G_BL_1MA"
-            ):
-                blend_method = "BLEND"
-            else:
-                blend_method = "OPAQUE"
-        else:
-            rendermode = f3dMat.rdp_settings.rendermode_preset_cycle_1
-            if not is_one_cycle:
-                rendermode = f3dMat.rdp_settings.rendermode_preset_cycle_2
-
-            f3d = get_F3D_GBI()
-            r_mode = getattr(f3d, rendermode, f3d.G_RM_AA_ZB_OPA_SURF)
-            if r_mode & f3d.CVG_X_ALPHA:
-                blend_method = "CLIP"
-            else:
-                cfunc = GBL_c1 if is_one_cycle else GBL_c2
-                xlu_comb = r_mode & cfunc(f3d.G_BL_CLR_IN, f3d.G_BL_A_IN, f3d.G_BL_CLR_MEM, f3d.G_BL_1MA)
-                if xlu_comb and r_mode & f3d.FORCE_BL:
-                    blend_method = "BLEND"
-                else:
-                    blend_method = "OPAQUE"
 
-    return blend_method
+def does_blender_use_mix(
+    settings: "RDPSettings",
+    mix: str,
+    default_for_no_rendermode: bool = False
+) -> bool:
+    if not settings.set_rendermode:
+        return default_for_no_rendermode
+    is_one_cycle = settings.g_mdsft_cycletype == "G_CYC_1CYCLE"
+    return settings.blend_b1 == mix or (is_one_cycle and settings.blend_b2 == mix)
+
+
+def is_blender_lerp(
```

Comment:
why "lerp"?
If because you can't think of a name, huh, maybe
```suggestion
def is_blender_cycle_equal(
```
since it checks if a blender cycle equals the arguments?

Reply:
So first of all, "lerp" is a programming term which means "linear interpolate", this is used by SGI in the N64 context as well as many other places.

I'm referring to the linear interpolation equation part of the blender, as opposed to the other settings within the rendermode. Now the other part of your question is why don't I call this "cycle". Other Fast64 code refers to the two of them as "cycle-dependent rendermode settings", so I guess the real name would be "the settings for one of the cycles within the cycle-dependent rendermode settings". To me "blender cycle" means the timestep within which the blender runs computations, not a subset of the settings for those computations.

With that said I agree the name is a bit unclear, do you have any other suggestions besides `is_blender_cycle_equal`? Maybe `is_blender_equation` or `is_blender_set_to_lerp`?

Reply:
"equation" would be an improvement over lerp yeah
I knew about lerp = linear interpolation, didn't make the link with the blender equation since it can be configured in ways that don't look like lerping (e.g. RM_ADD)

I like `is_blender_equation_equal` ? but if you don't then `is_blender_equation` is fine to me too

Reply:
Changed to `is_blender_equation_equal`